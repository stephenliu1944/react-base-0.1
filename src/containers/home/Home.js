import catPNG from 'images/cat.png';
import styles from './home.scss';
import SubComponent from './components/subComponent/SubComponent';
import { getIPInfo, uploadURL } from 'services/demo';

export default class Home {
    
    constructor() {
        this.sub = new SubComponent();
        /** 
         * HTTP Demo 
         */ 
        // http request
        getIPInfo('210.75.225.254').then((data) => {

        }, (error) => {

        });
        // create a prerequest object
        var obj = uploadURL({
            name: 'stephen',
            age: 35
        });
        // obj: { url, params, data, headers }
    }

    render() {
        return (`
            <div class=${styles.home}>
                <h1>Home</h1>
                <h1>我是容器组件</h1>
                <img src=${catPNG} /> 
                ${this.sub.render()}
            </div>
        `);
    }
}