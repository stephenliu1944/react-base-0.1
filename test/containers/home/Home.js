import Home from 'containers/home/Home';

it('check home.render()', () => {
    var home = new Home();
    expect(home.render()).toMatchSnapshot();
});

// TODO: