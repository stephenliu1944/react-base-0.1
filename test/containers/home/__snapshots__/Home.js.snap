// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check home.render() 1`] = `
"
            <div class=home>
                <h1>Home</h1>
                <h1>我是容器组件</h1>
                <img src=cat.png /> 
                
            <div class=subComponent>
                <h3>SubComponent</h3>
                <h3>我是子组件</h3>
            </div>
        
            </div>
        "
`;
